taichi/backends/metal/kernel_util.h

```diff
@@ -63,48 +63,82 @@ struct KernelAttributes {
   std::string debug_string() const;
 };
 
+// This class contains the attributes descriptors for both the input args and
+// the return values of a Taichi kernel.
+//
 // Note that all Metal kernels belonging to the same Taichi kernel will share
-// the same kernel args (attributes + Metal buffer). This is because kernel
-// arguments is a Taichi-level concept.
+// the same kernel args (i.e. they use the same Metal buffer for input args and
+// return values). This is because kernel arguments is a Taichi-level concept.
+//
+// TODO(#909): Rename to KernelArgsRetsAttributes
 class KernelArgsAttributes {
```

Comment:
Why not apply in this PR?

Reply:
I thought the renaming here might make it harder to review. But I'll do it since you don't object..

Reply:
Done